cupy_setup_build.py

```diff
@@ -480,6 +495,15 @@ def preconfigure_modules(compiler, settings):
                     settings['library_dirs'].append(lib_path)
                     break
 
+        if module['name'] == 'cusparselt':
+            cusparselt_path = os.environ.get('CUSPARSELT_PATH', '')
+            inc_path = os.path.join(cusparselt_path, 'include')
+            if os.path.exists(inc_path):
+                settings['include_dirs'].append(inc_path)
+            lib_path = os.path.join(cusparselt_path, 'lib64')
+            if os.path.exists(lib_path):
+                settings['library_dirs'].append(lib_path)
```

Comment:
One request: In order to work with Conda-Forge, could you also search for `$CUSPARSELT_PATH/lib` if it's not found  in `$CUSPARSELT_PATH/lib64`? Conda puts all the libraries under `$CONDA_PREFIX/lib`, so my plan for https://github.com/conda-forge/cupy-feedstock/issues/111 is to set `$CUSPARSELT_PATH=$CONDA_PREFIX`, which will work (verified locally).

Reply:
I think we don't have to introduce this env var.
`CUTENSOR_PATH` was needed because it has a special directory hierarchy (`lib/$CUDA_VERISON/libcutensor.*`), but cuSPARSELt seems not, so I think users can just put it to CFLAGS etc, like cuDNN/NCCL.

Reply:
Sorry, don't mind me. @kmaehashi's suggestion is nice to have, but regardless of what's done here, when building on Conda-Forge the `cusparselt` package can be found without problem.

Reply:
Thanks for your suggestions.

Indeed, if you set the environment variables CFLAGS, LDFLAGS, and LD_LIBRARY_PATH appropriately, you don't need to bother adding the environment variable CUSPARSELT_PATH. I will remove the CUSPARSELT_PATH environment variable that I added to cupy_setup_build.py.

    $ export CUSPARSELT=/path/to/the/libcutensorlt
    $ CFLAGS="-I${CUSPARSELT}/include" LDFLAGS="-L${CUSPARSELT}/lib64" python setup.py install
    ...
    $ cd examples/cusparselt
    $ LD_LIBRARY_PATH=${CUSPARSELT}/lib64:${LD_LIBRARY_PATH} python matmul.py
     ** On entry to cusparseLtInit(): cusparseLt currently supports only SM80 (current: SM70)
    
    Traceback (most recent call last):
      File "matmul.py", line 29, in <module>
        cusparselt.init(handle)
      File "cupy_backends/cuda/libs/cusparselt.pyx", line 202, in cupy_backends.cuda.libs.cusparselt.init
      File "cupy_backends/cuda/libs/cusparselt.pyx", line 205, in cupy_backends.cuda.libs.cusparselt.init
      File "cupy_backends/cuda/libs/cusparselt.pyx", line 196, in cupy_backends.cuda.libs.cusparselt.check_status
cupy_backends.cuda.libs.cusparse.CuSparseError: CUSPARSE_STATUS_ARCH_MISMATCH

Reply:
> but cuSPARSELt itself seems to only support CC=80

Well, cuSPARSELt targets GPUs with TensorCore units that supports structured sparsity 😅

Reply:
> > but cuSPARSELt itself seems to only support CC=80
> 
> Well, cuSPARSELt targets GPUs with TensorCore units that supports structured sparsity 😅

Sounds like it's not possible to support older SMs, right?

Reply:
@leofang You are correct, there is no 'fallback' on older hardware.